File: src-tauri/src/models/equipment.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, sqlx::FromRow, Serialize, Deserialize)]
pub struct Equipment {
    pub id: i32,
    pub service_tag: String,
    pub name: String,
    pub description: String,
    pub location_id: i32,
    pub created_at: String,
    pub updated_at: String,
}

File: src/domain/models/equipment.ts
export type Equipment = {
	id: string
	service_tag: string
	name: string
	description: string
	location_id: string
	created_at: string
	updated_at: string
}

export type EquipmentWithLocation = Equipment & {
	location_name: string
}

export type EquipmentLocation = {
	id: string
	name: string
	description: string
	created_at: string
	updated_at: string
}
